base/sparse/sparsematrix.jl

```diff
@@ -292,7 +292,10 @@ function copy!(A::SparseMatrixCSC, B::SparseMatrixCSC)
     return A
 end
 
-similar(S::SparseMatrixCSC, Tv::Type = eltype(S)) = SparseMatrixCSC(S.m, S.n, copy(S.colptr), copy(S.rowval), Vector{Tv}(length(S.nzval)))
+function similar(S::SparseMatrixCSC, ::Type{Tv} = eltype(S)) where {Tv}
```

Comment:
Editing in github is not going to be convenient for contributing to Julia. Git can be a bit difficult at first, but you can pretty easily get into the groove. Basically, once you have a branch set up to be a PR, you can keep committing to the same branch. When we merge your PR, we will combine into one big commit.

Opening a new PR every time creates a lot of noise otherwise, and splits the discussion and reviews into too many places.